### DISCUSSION TITLE
Question re: trivy k8s cluster scan behaviour

### DISCUSSION THREAD

**Discussion by dstuartkelly** [aquasecurity/trivy#5054] [Q&A] [answered]
### Question

I am interested in scanning for vulnerabilities in a number of pods (approx 250 - 350) running in my k8s cluster.

Up till now, I have been generating a unique list of my images using `kubectl` then running `trivy image --scanners vuln $IMAGE` for each image, outputting via a html template and appending all the outputs to a giant html file (approx 6MB). Its time consuming but it has worked for its purpose.

I recently discovered `trivy k8s`  and the summary report seems to be very close to what I need to do. I am running the command as follows
`trivy k8s cluster --scanners=vuln --components=workload --report=summary --timeout 30m0s`
It is completing with a few errors that I am expecting (Permission denied when images are in a private registry) but I have a question about how the container scanning is handled in the case of the `trivy k8s cluster`?

Does `trivy k8s` interact directly with the running container to scan it or does it download and cache a copy of the running containers image?

Thank you!

### Target

Kubernetes

### Scanner

Vulnerability

### Output Format

Table

### Mode

Standalone

### Operating System

ubuntu

### Version

```bash
Version: 0.44.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-08-28 06:09:31.478888986 +0000 UTC
  NextUpdate: 2023-08-28 12:09:31.478888586 +0000 UTC
  DownloadedAt: 2023-08-28 17:05:41.831006154 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2023-08-28 00:55:19.060540828 +0000 UTC
  NextUpdate: 2023-08-31 00:55:19.060540128 +0000 UTC
  DownloadedAt: 2023-08-28 12:28:11.509774773 +0000 UTC
Policy Bundle:
  Digest: sha256:2e95a2d5d45de8ebecae53a97403230a6c608a579b082f3de170f3cf09e46243
  DownloadedAt: 2023-08-25 16:49:07.960443716 +0000 UTC
```

**Comment by knqyf263** [comment]
@chen-keinan

**Comment by chen-keinan** [comment, marked as answer]
@dstuartkelly `trivy k8s ...` use `k8s go client` behind the scene which is the equivalent for `kubectl` it fetch all resources from the cluster and download the images for each resource and scan it

**Reply by dstuartkelly** [reply]
Thank you for the response @chen-keinan 

Just to confirm my understanding, 
`trivy k8s ...` queries the clusters API directly to discover what images are in use. It will then download the images from the relevant repositories and the scanning takes place locally on the client machine where the `trivy k8s ...` command was run.

**Reply by chen-keinan** [reply]
yes exactly
